增加最后一个索引条目存储了上图中颗粒1082的主键列的最大值与图片不符的描述

iammagicc opened the pull request:
为什么图中显示的数据与结论不符合？因为图中的数据是禁用了自适应索引粒度后得到的，默认情况下索引粒度是自适应的。 https://clickhouse.com/docs/en/optimize/sparse-primary-indexes We mentioned in the beginning of this guide in the "DDL Statement Details", that we disabled adaptive index granularity (in order to simplify the discussions in this guide, as well as make the diagrams and results reproducible). For tables with adaptive index granularity (index granularity is adaptive by default) the size of some granules can be less than 8192 rows depending on the row data sizes. 我们在本指南开头的“DDL 语句详细信息”中提到，我们禁用了自适应索引粒度（为了简化本指南中的讨论，并使图表和结果可重现）。 对于具有自适应索引粒度的表（默认情况下索引粒度是自适应的），某些粒度的大小可以小于 8192 行，具体取决于行数据大小。


https://clickhouse.com/docs/en/whats-new/changelog/2019#experimental-features-1 ClickHouse Release 19.6.3.18, 2019-06-13
Experimental Features：实验性特性   
Add setting index_granularity_bytes (adaptive index granularity) for MergeTree* tables family. 为合并树家族的表系列添加设置index_granularity_bytes（自适应索引粒度）。

ClickHouse Release 19.10.1.5, 2019-07-12
Performance Improvement：优化改进
Add the possibility to write the final mark at the end of MergeTree columns. It allows to avoid useless reads for keys that are out of table data range. It is enabled only if adaptive index granularity is in use. 添加在合并树列末尾写入最终标记的可能性。它允许避免对超出表数据范围的键进行无用的读取。仅当使用自适应索引粒度时，才会启用它。

<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/run_check.py
-->
### Changelog category (leave one):
- Documentation (changelog entry is not required)


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->


> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

iammagicc committed:
增加最后一个索引条目存储了上图中颗粒1082的主键列的最大值与图片不符的描述

为什么图中显示的数据与结论不符合？因为图中的数据是禁用了自适应索引粒度后得到的，默认情况下索引粒度是自适应的。
https://clickhouse.com/docs/en/optimize/sparse-primary-indexes
We mentioned in the beginning of this guide in the "DDL Statement Details", that we disabled adaptive index granularity (in order to simplify the discussions in this guide, as well as make the diagrams and results reproducible).
For tables with adaptive index granularity (index granularity is adaptive by default) the size of some granules can be less than 8192 rows depending on the row data sizes.
我们在本指南开头的“DDL 语句详细信息”中提到，我们禁用了自适应索引粒度（为了简化本指南中的讨论，并使图表和结果可重现）。
对于具有自适应索引粒度的表（默认情况下索引粒度是自适应的），某些粒度的大小可以小于 8192 行，具体取决于行数据大小。


https://clickhouse.com/docs/en/whats-new/changelog/2019#experimental-features-1
ClickHouse Release 19.6.3.18, 2019-06-13
Experimental Features：实验性特性   
Add setting index_granularity_bytes (adaptive index granularity) for MergeTree* tables family.
为合并树家族的表系列添加设置index_granularity_bytes（自适应索引粒度）。

ClickHouse Release 19.10.1.5, 2019-07-12
Performance Improvement：优化改进
Add the possibility to write the final mark at the end of MergeTree columns. It allows to avoid useless reads for keys that are out of table data range. It is enabled only if adaptive index granularity is in use.
添加在合并树列末尾写入最终标记的可能性。它允许避免对超出表数据范围的键进行无用的读取。仅当使用自适应索引粒度时，才会启用它。

iammagicc committed:
Merge branch 'master' into patch-2

hanfei1991 commented on docs/zh/guides/improving-query-performance/sparse-primary-indexes.md:
```diff
@@ -370,6 +370,25 @@ UserID.bin，URL.bin，和EventTime.bin是<font face = "monospace">UserID</font>
+- https://clickhouse.com/docs/en/optimize/sparse-primary-indexes
+- We mentioned in the beginning of this guide in the "DDL Statement Details", that we disabled adaptive index granularity (in order to simplify the discussions in this guide, as well as make the diagrams and results reproducible).
+- For tables with adaptive index granularity (index granularity is adaptive by default) the size of some granules can be less than 8192 rows depending on the row data sizes.
+- 我们在本指南开头的“DDL 语句详细信息”中提到，我们禁用了自适应索引粒度（为了简化本指南中的讨论，并使图表和结果可重现）。
```
seems you should put this to the above note (line 340)

iammagicc replied:
Yes,i think it's a good idea,this has already been mentioned above.Maybe I should add a new note behind line 340 for readers to understand better?

iammagicc replied:
OK,I'll remove the quoat English text and think carefully about the translation.

iammagicc committed:
image errors and adaptive index granularity

Add a description of the relationship between image errors and adaptive index granularity.And create a new node with this description.

iammagicc committed:
Merge branch 'master' into patch-2

iammagicc committed:
delete quoat English text，change the translation

As the title says，I delete quoat English text.The Chinese translation has been modified to make it easier for readers to understand.

iammagicc committed:
Merge branch 'master' into patch-2

iammagicc commented on docs/zh/guides/improving-query-performance/sparse-primary-indexes.md:
```diff
@@ -335,6 +335,25 @@ UserID.bin，URL.bin，和EventTime.bin是<font face = "monospace">UserID</font>
 第一个(根据磁盘上的物理顺序)8192行(它们的列值)在逻辑上属于颗粒0，然后下一个8192行(它们的列值)属于颗粒1，以此类推。
 
+:::note
+- 为什么上面和接下来的图片中显示的橙色数据与结论（1.最后一个颗粒我们标记的是最大的值；2.索引条目(索引标记)不是基于表中的特定行，而是基于颗粒。）不符合？
```
这段话是我自己加的。。不然很突兀的把下面的贴上去吗

hanfei1991 replied:
如果觉得文档表达的不清楚，应该先在英文文档里提交对应的英文解释，否则中文文档应该直译。
我的意思是直接把漏译的部分原封不动的加到下一个 note ☺️

iammagicc replied:
英文文档中没有我说的这些，当时中文文档本来就添加了如最后一个颗粒使用最终索引这些问题，如果不加我所添加的注解，读者可能看不懂。

hanfei1991 replied:
我作为读者觉得原文表述的更好哈，Sorry...

不过很感谢你的 PR，翻译得很好！
